### ISSUE TITLE
tis calculation is wrong when applying calculate_per_token_loss

### ISSUE THREAD

**Issue by yyht** [THUDM/slime#364]
```
def loss_function(args, batch, num_microbatches, logits):
    num_tokens = sum(batch["response_lengths"])
    num_samples = len(batch["response_lengths"])

```
in megatron_utils/loss.py, when applying calculate_per_token_loss, the 
```
num_tokens
```
should also consider the loss-masks.

Otherwise, the train/tis will be lower by the ratio of aw_response_length/response_length_mean and wo_obs_response_length/response_length_mean.

The results could be misleading since train/tis measure the difference between training-backend and rollout-backend.

**Comment by zhuzilin** [comment]
Nice catch! Fixed in #365
